### ISSUE TITLE
applePayConfiguration Typescript object doesn't match docs

### ISSUE THREAD

**Issue by Kaschman** [Adyen/adyen-web#2074]
**Describe the bug**
When following [the guide](https://docs.adyen.com/payment-methods/apple-pay/web-component) for adding ApplePay using Adyen's certificate with Web Components, the `applePayConfiguration` object described in the docs does not compile. It requires a `configuration` object.

**To Reproduce**
1. Create an applePayConfiguration object as described in the guide in a Typescript repo
```
const applePayConfiguration = {
--
amount: {
value: 1000,
currency: "EUR"
},
countryCode: "DE"
};
```
2. Pass the configuration to the component creation method:
```
const applePayComponent = adyenCheckout.create<'applepay'>(
          'applepay',
          applePayConfiguration,
        )
```
3. The code will not compile

**Additional context**
I believe the fix is a simple update to have the `configuration` property on this object be optional. The current behavior is confusing and leads the user down the wrong path - for instance passing an empty configuration object, which breaks the default behavior of using Adyen's cert.

I was able to fix in my code by casting `applePayConfiguration` to `any` and not passing a `configuration` property as part of `applePayConfiguration`

**Comment by ribeiroguilherme** [comment]
Hi @Kaschman ,

Indeed, the `configuration` object comes from the Backend, therefore it is optional in the UI. I believe that the `amount` should be optional as well (or not even required actually), since it is passed to the main `AdyenCheckout` instance when it is created. 

We will take a better look into it and will keep you posted about the fix.

**Comment by ribeiroguilherme** [comment]
@Kaschman We release the fix on version 5.39.0 . Cheers
